subprojects/core/src/main/java/org/gradle/execution/plan/DefaultExecutionPlan.java

```diff
@@ -595,36 +596,54 @@ private MutationInfo getResolvedMutationInfo(Node node) {
     private void resolveMutations(final MutationInfo mutations, Node node) {
         if (node instanceof LocalTaskNode) {
             final LocalTaskNode taskNode = (LocalTaskNode) node;
-            TaskInternal task = taskNode.getTask();
+            final TaskInternal task = taskNode.getTask();
             ProjectInternal project = (ProjectInternal) task.getProject();
             ServiceRegistry serviceRegistry = project.getServices();
             final FileResolver resolver = serviceRegistry.get(FileResolver.class);
             PropertyWalker propertyWalker = serviceRegistry.get(PropertyWalker.class);
             try {
                 TaskPropertyUtils.visitProperties(propertyWalker, task, new PropertyVisitor.Adapter() {
                     @Override
-                    public void visitOutputFileProperty(String propertyName, boolean optional, PropertyValue value, OutputFilePropertyType filePropertyType) {
-                        addCanonicalizedPaths(
+                    public void visitOutputFileProperty(final String propertyName, boolean optional, final PropertyValue value, final OutputFilePropertyType filePropertyType) {
+                        reportDeadlockOnParameterResolution(
```

Comment:
This method name sounds like we already know there will be a deadlock. It should be more like something `withDeadlockHandling()` or something. WDYT?

Reply:
Renamed to `withDeadlockHandling`.